packages/react-vapor/src/components/unsavedChangeModal/test/UnsavedChangeModalProvider.spec.tsx

```diff
@@ -0,0 +1,155 @@
+                        />
+                    );
+                }}
+            </UnsavedChangesModalProvider>
+        );
+
+    const heavilyMountUnsavedModalProvider = (isDirty: boolean = false) =>
+        mount(
+            <UnsavedChangesModalProvider isDirty={isDirty}>
+                {({promptBefore}) => {
+                    return (
+                        <Provider store={store}>
+                            <ModalCompositeConnected
+                                id="👾"
+                                title="🙄"
+                                classes={['mod-slide-in-bottom', 'mod-big', 'mod-stick-bottom']}
+                                modalBodyChildren="🙈 🙉 🙊"
+                                modalFooterChildren={
+                                    <button
+                                        className="btn"
+                                        onClick={() =>
+                                            promptBefore(promptBeforeClickActionSpy) && regularClickActionSpy()
+                                        }
+                                    />
+                                }
+                                modalBodyClasses={['mod-header-padding', 'mod-form-top-bottom-padding']}
+                                docLink={{url: 'https://www.coveo.com', tooltip: {title: 'Go to coveo.com'}}}
+                            />
+                        </Provider>
+                    );
+                }}
+            </UnsavedChangesModalProvider>
+        );
+
+    it('should mount and unmount without trowing', () => {
+        expect(() => {
+            shallowMountUnsavedModalProvider().unmount();
+        }).not.toThrow();
+    });
+
+    describe('when dirty changes', () => {
+        it('should open the UnsavedChangesModal to interrupt the leaving action', () => {
+            heavyUnsavedChangesModalProvider = heavilyMountUnsavedModalProvider(true);
+            expect(heavyUnsavedChangesModalProvider.childAt(1).prop('isOpen')).toBeFalsy();
+
+            const heavyModal: any = heavyUnsavedChangesModalProvider.find(ModalCompositeConnected);
+            heavyModal.props().modalFooterChildren.props.onClick();
+            heavyUnsavedChangesModalProvider.update();
+            expect(heavyUnsavedChangesModalProvider.childAt(1).prop('isOpen')).toBeTruthy();
+        });
+        describe('PromptBefore function', () => {
+            it('should return false', () => {
+                unsavedChangesModalProvider = shallowMountUnsavedModalProvider(true);
+                unsavedChangesModalProvider
+                    .childAt(0)
+                    .props()
+                    .modalFooterChildren.props.onClick();
+                expect(regularClickActionSpy).not.toHaveBeenCalled();
+            });
+        });
+        it('should use the callBack function in the promptBefore function when confirming the exit', () => {
```

Comment:
detail, but `callBack` => `callback` and a couple of other tests also has that

Reply:
:eyes:  thanks. That is a pretty big detail in fact.